packages/hierarchy-builder/src/test/internal/operators/grouping/BaseClassGrouping.test.ts

```diff
@@ -0,0 +1,253 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { from } from "rxjs";
+import { HierarchyNode } from "../../../../hierarchy-builder/HierarchyNode";
+import { createGroupingOperator } from "../../../../hierarchy-builder/internal/operators/Grouping";
+import { IMetadataProvider } from "../../../../hierarchy-builder/Metadata";
+import { createGetClassStub, createTestNode, getObservableResult, isMock, TStubClassFunc } from "../../../Utils";
+
+describe("BaseClassGrouping", () => {
+  const metadataProvider = {} as unknown as IMetadataProvider;
+  let stubClass: TStubClassFunc;
+  beforeEach(() => {
+    stubClass = createGetClassStub(metadataProvider).stubClass;
+  });
+
+  it("does not group non-instance nodes", async () => {
+    const nodes: HierarchyNode[] = [
+      {
+        label: "custom",
+        key: "test",
+        children: false,
+        params: {
+          grouping: {
+            byBaseClasses: {
+              fullClassNames: ["CustomSchema.CustomClass"],
+            },
+          },
+        },
+      },
+    ];
+    const result = await getObservableResult(from(nodes).pipe(createGroupingOperator(metadataProvider)));
+    expect(result).to.deep.eq(nodes);
+  });
+
+  it("does not group if not entity or relationship baseClass", async () => {
+    const nodes = [
+      createTestNode({
+        key: { type: "instances", instanceKeys: [{ className: "TestSchema.TestClass", id: "0x1" }] },
+        params: {
+          grouping: {
+            byBaseClasses: {
+              fullClassNames: ["TestSchema.TestParentClass"],
+            },
+          },
+        },
+      }),
+    ];
+    stubClass({ schemaName: "TestSchema", className: "TestClass", is: isMock });
+    stubClass({
+      schemaName: "TestSchema",
+      className: "TestParentClass",
+      classLabel: "TestSchema.TestParentClass",
```

Comment:
The label property should be in label format rather than just set to full class name. However, I don't think you need it at all in this test.

Reply:
Removed